File: src/Undersoft.SharedServiceCenter.Web.Application.React/src/undersoft.ssc.web.application.client/src/account/accountIdentity.ts
import { AccountIdentityClaim } from "./accountIdentityClaim";
import { AccountIdentityRole } from "./accountIdentityRole";
import { IdentityUser } from "./identity/identityUser";
import { Credentials } from "./identity/credentials";
import { Model } from "@/models/model";

export interface AccountIdentity extends Model {
    Info?: IdentityUser;  
    Roles?: AccountIdentityRole;
    Claims?: AccountIdentityClaim;
    Credentials?: Credentials;
}

File: src/Undersoft.SoftwareDevelopmentKit/src/Undersoft.SDK.Service.Application/Account/Identity/AccountIdentity.cs
using Microsoft.AspNetCore.Identity;
using System.Security.Claims;
using Undersoft.SDK.Security.Identity;
using Undersoft.SDK.Service.Data.Contract;

namespace Undersoft.SDK.Service.Application.Account.Identity;

public class AccountIdentity : DataObject, IContract, IAccountIdentity<long>
{
    public IdentityUser<long> Info { get; set; } = new IdentityUser<long>() { Id = Unique.NewId };

    public Registry<AccountIdentityRole<long>> Roles { get; set; }

    public Registry<AccountIdentityClaim<long>> Claims { get; set; }

    public IEnumerable<Claim> GetClaims() { return Claims.Select(c => c.Claim); }

    public ICredentials Credentials { get; set; } = new Credentials();

    public AuthorizationNotes Notes { get; set; } = new AuthorizationNotes();

    public bool Authorized { get; set; }

    public bool Authenticated { get; set; }
}
